### PR TITLE
Add pieces sti

### PR DESCRIPTION
**Pull request by zenkbaries**
Added Single Table Inheritance for Pieces for the following pieces:
Pawn
Knight
Bishop
Rook
Queen
King

and added scopes to Pieces Models for each pieces

### CODE REVIEW

File: app/models/piece.rb [hunk 1]
```diff
@@ -1,4 +1,11 @@
 class Piece < ActiveRecord::Base
-    belongs_to :player
-    belongs_to :game    
+  belongs_to :player
+  belongs_to :game
 end
+
+scope :pawns, -> { where(race: 'Pawn') }
```
Comment (mscoutermarsh) [inline app/models/piece.rb]: Scopes would have to be within the class (see line #4). Above the end

**Comment by mscoutermarsh** [comment]
Almost done!

To finish this, you'll need to add a `type` column to `Piece`.

In the guide I sent you, they use `race`. Which I'm now realizing is confusing. Ignore that and use `type` instead.

`type` is the default that Rails uses for single table inheritance.

File: app/models/piece.rb [hunk 2]
```diff
@@ -1,4 +1,11 @@
...
+scope :knights, -> { where(race: 'Knight') }
+scope :bishops, -> { where(race: 'Bishop') }
+scope :rooks, -> { where(race: 'Rook') }
+scope :queens, -> { where(race: 'Queen') }
+scope :kings, -> { where(race: 'King') }
```
Comment (mscoutermarsh) [inline app/models/piece.rb]: instead of `race` let's use `type`

**Comment by zenkbaries** [comment]
Ok I replaced all 'race' with 'type'
and the type column is already in pieces from last week migration file.

I had a conflict and resolved (seems that pieces from master file is different from add-pieces-sti file which i know I change but they call it 'conflict'  i dont know why) so i fixed it and now it merge successfully  (not realized that I should have asked @mscoutermarsh for review before 'merge' it)

**Pull request merged**